### PR TITLE
Add script to cut release branch

### PR DESCRIPTION
**Pull request by guangy10**
Step 1 of release process: Create a release branch.

The branch is named with `release/MAJOR.MINOR`.

From executorch root, run `./scripts/release/cut-release-branch.sh` will cut a release branch off stable branch (`viable/strict`) by default. Use `GIT_BRANCH_TO_CUT_FROM=main ./scripts/release/cut-release-branch.sh` to cut off `main` branch.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #2578
* __->__ #2567

Differential Revision: [D55208762](https://our.internmc.facebook.com/intern/diff/D55208762)

### CODE REVIEW

File: scripts/release/cut-release-branch.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+'
+
+set -eou pipefail
+
+GIT_TOP_DIR=$(git rev-parse --show-toplevel)
+GIT_REMOTE=${GIT_REMOTE:-origin}
+GIT_BRANCH_TO_CUT_FROM=${GIT_BRANCH_TO_CUT_FROM:-viable/strict}
+
+# should output something like 1.11
+RELEASE_VERSION=${RELEASE_VERSION:-$(cut -d'.' -f1-2 "${GIT_TOP_DIR}/version.txt")}
```
Comment (guangy10) [inline scripts/release/cut-release-branch.sh:23]: Do we want to name the release branch with the full version or just the MAJOR.MINOR version?

File: version.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+0.2.0-rc1
```
Comment (huydhn) [inline version.txt:1]: Having `rc1` suffix here is a bit out of place, but it depends on how the build works in ET.  Here is how we implement it on PyTorch and other libraries atm:

1. The version on `main` points to the next release, i.e. https://github.com/pytorch/pytorch/blob/main/version.txt is pointing to `2.4.0a0`.  This is used to built nightly binaries, i.e. https://download.pytorch.org/whl/nightly/cpu-cxx11-abi/torch-2.4.0.dev20240316%2Bcpu.cxx11.abi-cp310-cp310-linux_x86_64.whl
2. After doing branch cut, we bump the version, i.e. https://github.com/pytorch/pytorch/pull/121974
3. Realese binaries are built without the RC suffix, so the latest RC binaries will overwrite previous one, and the test  installation is always the same throughout the release process, i.e. `pip3 install torch torchvision torchaudio executorch==0.2.0 --index-url https://download.pytorch.org/whl/test/cpu`
4. Once the final RC is ready, the files under https://download.pytorch.org/whl/test/cpu will be published to pypi/conda/poetry.

So, in a sense, RC tag is used to triggered release build, but not included as part of the artifact name.

File: scripts/release/cut-release-branch.sh (hunk 1 above)
Comment (huydhn) [reply scripts/release/cut-release-branch.sh:23]: In PyTorch, the release branch name is just `release/MAJOR.MINOR`, for example `release/2.2`, which is corresponding to the major release `2.2.0`.  Then, we have patch release, i.e. `2.2.1`, 2.2.2` after that to fix bugs that people report on that release branch.  The timeline for a major PyTorch release is every 3 month, so there are 2 patch release in between before the next major release.  Only major release requires doing branch cut, `release/2.3` is next.

**Review by huydhn — APPROVED** [review]

**Comment by guangy10** [comment]
@guangy10 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D55208762).

**Pull request closed without merging**